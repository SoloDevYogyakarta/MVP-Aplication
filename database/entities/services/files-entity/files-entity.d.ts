import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class FileEntity extends BaseEntity {
    originalname: string;
    filepath: string;
    type: string;
    desc: string;
    browse: string;
    order_id: number;
}
export type FileCreationAttribute = Optional<FileEntity, 'id'>;
export interface FileInstance extends Model<FileCreationAttribute, FileEntity>, FileEntity {
}
export declare const fileEntity: import("sequelize").ModelCtor<FileInstance>;
