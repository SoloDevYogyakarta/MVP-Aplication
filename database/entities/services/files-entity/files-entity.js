"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fileEntity = exports.FileEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class FileEntity extends base_entity_1.BaseEntity {
}
exports.FileEntity = FileEntity;
exports.fileEntity = entity_1.sequelize.define('FILES', {
    originalname: sequelize_1.DataTypes.STRING,
    filepath: sequelize_1.DataTypes.STRING,
    type: sequelize_1.DataTypes.STRING,
    order_id: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'FILES',
    schema: 'PRODUCTS',
    timestamps: false,
});
//# sourceMappingURL=files-entity.js.map