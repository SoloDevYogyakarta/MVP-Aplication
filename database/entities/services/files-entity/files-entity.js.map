{"version":3,"file":"files-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/services/files-entity/files-entity.ts"],"names":[],"mappings":";;;AAAA,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,UAAW,SAAQ,wBAAU;CAOzC;AAPD,gCAOC;AAMY,QAAA,UAAU,GAAG,kBAAS,CAAC,MAAM,CACxC,OAAO,EACP;IACE,YAAY,EAAE,qBAAS,CAAC,MAAM;IAC9B,QAAQ,EAAE,qBAAS,CAAC,MAAM;IAC1B,IAAI,EAAE,qBAAS,CAAC,MAAM;IACtB,IAAI,EAAE,qBAAS,CAAC,IAAI;IACpB,MAAM,EAAE,qBAAS,CAAC,MAAM;IACxB,QAAQ,EAAE,qBAAS,CAAC,OAAO;CACK,EAClC;IACE,SAAS,EAAE,OAAO;IAClB,MAAM,EAAE,UAAU;IAClB,UAAU,EAAE,KAAK;CAClB,CACF,CAAC"}