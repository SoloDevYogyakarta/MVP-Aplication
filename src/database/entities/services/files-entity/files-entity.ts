import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class FileEntity extends BaseEntity {
  originalname: string;
  filepath: string;
  type: string;
  desc: string;
  browse: string;
  order_id: number;
}

export type FileCreationAttribute = Optional<FileEntity, 'id'>;
export interface FileInstance
  extends Model<FileCreationAttribute, FileEntity>,
    FileEntity {}
export const fileEntity = sequelize.define<FileInstance>(
  'FILES',
  {
    originalname: DataTypes.STRING,
    filepath: DataTypes.STRING,
    type: DataTypes.STRING,
    desc: DataTypes.TEXT,
    browse: DataTypes.STRING,
    order_id: DataTypes.INTEGER,
  } as ModelAttributes<FileInstance>,
  {
    tableName: 'FILES',
    schema: 'PRODUCTS',
    timestamps: false,
  },
);
