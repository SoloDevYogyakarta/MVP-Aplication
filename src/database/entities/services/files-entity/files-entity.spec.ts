import { ModelCtor } from 'sequelize';
import { fileEntity, FileInstance } from './files-entity';

describe('FileEntity', () => {
  let entity: ModelCtor<FileInstance>;

  beforeEach(() => {
    entity = fileEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());
});
