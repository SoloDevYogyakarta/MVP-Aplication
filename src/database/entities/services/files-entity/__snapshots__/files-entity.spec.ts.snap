// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileEntity render correctly 1`] = `[Function]`;
